### PR TITLE
Fix "pending" parameter on RPC block requests

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

File: ethcore/src/block.rs [hunk 1]
```diff
@@ -41,8 +41,18 @@ impl Block {
...
 	}
+
+	/// Get the RLP-encoding of the block without the seal.
+	pub fn rlp_bytes(&self) -> Bytes {
+		let mut block_rlp = RlpStream::new_list(3);
+		self.header.stream_rlp(&mut block_rlp, Seal::Without);
+		block_rlp.append(&self.transactions);
+		block_rlp.append(&self.uncles);
+		block_rlp.out()
+	}
```
Comment (rphmeier) [inline ethcore/src/block.rs]: should take seal parameter. i would find this useful for minor snapshotting cleanup

File: ethcore/src/client/client.rs [hunk 2]
```diff
@@ -556,7 +564,7 @@ impl Client {
 			BlockID::Number(number) => Some(number),
 			BlockID::Hash(ref hash) => self.chain.block_number(hash),
 			BlockID::Earliest => Some(0),
-			BlockID::Latest => Some(self.chain.best_block_number())
+			BlockID::Latest | BlockID::Pending => Some(self.chain.best_block_number()),
```
Comment (rphmeier) [inline ethcore/src/client/client.rs]: pending would be best_block_number + 1?

Comment (gavofyork) [inline ethcore/src/client/client.rs]: would be were it to exist. if we get to here we assume pending isn't available, in which case we return the latest.

Comment (rphmeier) [inline ethcore/src/client/client.rs]: why add the pending variant to `BlockID` at all then? 
RPC requests could handle miner/client split and then client doesn't even have to know that it exists

File: ethcore/src/client/client.rs [hunk 3]
```diff
@@ -461,9 +461,17 @@ impl Client {
 	/// This will not fail if given BlockID::Latest.
 	/// Otherwise, this can fail (but may not) if the DB prunes state.
 	pub fn state_at(&self, id: BlockID) -> Option<State> {
+		// fast path for pending state.
```
Comment (tomusdrw) [inline ethcore/src/client/client.rs]:
```
match id.clone() {
  BlockID:Pending => self.miner.pending_state().or_else(|| Some(self.state())),
  BlockID::Latest => Some(self.state()),
  _ => {},
}
```
?

File: rpc/src/v1/types/block_number.rs [hunk 4]
```diff
@@ -64,7 +64,8 @@ impl Into<BlockID> for BlockNumber {
 			BlockNumber::Num(n) => BlockID::Number(n),
 			BlockNumber::Earliest => BlockID::Earliest,
 			// TODO: change this once blockid support pendingst,
```
Comment (tomusdrw) [inline rpc/src/v1/types/block_number.rs]: To remove?

File: ethcore/src/client/client.rs (hunk 2 above)
Comment (gavofyork) [inline ethcore/src/client/client.rs]: client has more immediate access to miner (through an `Arc`) and already manages other splits.

File: ethcore/src/client/client.rs (hunk 3 above)
Comment (gavofyork) [inline ethcore/src/client/client.rs]: missing two `return`s, but yeah - that should work

**Pull request merged**